contents/swift/transition-to-swift.md

```diff
@@ -0,0 +1,340 @@
+
+<!--
+If you have decided to build an app for your MacBook or iPhone (or any other Apple Product) or transitioning from another language to Swift, I am here to make your life easier. The common solutions to "How to learn Swift" from Quora and StackOverflow are to
+
+1. Dive into a project
+
+2. Read the official documentation
+https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/index.html
+
+
+Those are great suggestions but diving into a new project with no prior experience to Swift makes it an ardous trial of google searches of how to implement stuff. And reading documentation, while really helpful, might be overkill and boring as there might be many concepts that you already know.
+
+To ease this pain, I am providing a 3rd solution where I try to summarise the more common and important aspects of Swift that are generally different from other languages.
+-->
+
+# Getting Started
+
+## Good readings on Swift:
+- Design of Swift, why the change from obj-C:
+http://swiftjester.org/papers/whats-different-about-swift.html
+
+
+- Main Swift Concepts (less indepth, wide range of topics, long explanations):
+https://guides.codepath.com/ios/Swift-Basics
+
+- Basics of Apple Swift in playground:
+https://github.com/Codeido/swift-basics
+
+- Guide to Xcode(IDE for Swift):
+http://codewithchris.com/xcode-tutorial/
+
+- Another compilation of links:
+http://www.whoishostingthis.com/resources/swift/
+
+## Cheatsheets:
+- One page cheatsheet:
+https://koenig-media.raywenderlich.com/uploads/2014/06/RW-Swift-Cheatsheet-0_7.pdf
+
+- Multiple pages of cheatsheet (mainly code examples with comments):
+https://mhm5000.gitbooks.io/swift-cheat-sheet/content/index.html
+
+
+
+
+# Swift Basics
+
+## [The Absolute Basics](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/TheBasics.html)
+
+### Differences between Java and Swift:
+
+- let is similiar to final in Java
+
+- Unlike Java, it is not compulsory to annotate the type of a variable.
+
+  ```
+  var c: Int
+  var d //this is legal
+  ```
+
+- Instead of `for(int i = 0; i<100; i++) {` in Java, you would do `for i in 0..<100 {`.
+
+- nil in Swift is the same as null in Java
+
+### Things to note:
+
+- Constant and variable names can contain almost any character, including Unicode characters
+
+- There is the idea of mutability of variables. If a variable should not be immutable, use let to declare it as a immutable constant.
+
+  ```
+  let a = 2
+  var b = 3
+  ```
+
+- Once the type of a variable is decided, it cannot be changed.
+
+  ```
+  var c: Int
+  ```
+
+- There will be more readings on Error-Handling and Optionals.
```

Comment:
My plan is to cover this in the future and then put a link. Maybe I comment this out first?